federation/pkg/kubefed/util/util.go

```diff
@@ -285,27 +288,47 @@ func getRBACVersion(discoveryclient discovery.CachedDiscoveryInterface) (*schema
 		return nil, fmt.Errorf("Couldn't get clientset to create RBAC roles in the host cluster: %v", err)
 	}
 
+	// These are the RBAC versions we can speak
+	knownVersions := map[schema.GroupVersion]bool{
```

Comment:
This is a generic problem we need to solve for all API resources that kubefed creates.
Filed https://github.com/kubernetes/kubernetes/issues/50540